docs/designs/master_redesign.md

```diff
@@ -0,0 +1,136 @@
+# Master Design
+
+This document deseribes an ElasticDL master design.
```

Comment:
This document deseribes an ElasticDL master design. -> This document is a new design of the ElasticDL master.